lib/rubocop/cop/rails/lexically_scope_action_filter.rb

```diff
@@ -0,0 +1,117 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks that methods specified in the filter's `only`
+      # or `except` options are explicitly defined in the controller.
+      #
+      # You can specify methods of superclass or methods added by mixins
+      # on the filter, but these confuse developers. If you specify methods
+      # where are defined on another controller, you should define the filter
+      # in that controller.
+      #
+      # @example
+      #   # bad
+      #   class LoginController < ApplicationController
+      #     before_action :require_login, only: %i[index settings logout]
+      #
+      #     def index
+      #     end
+      #   end
+      #
+      #   # good
+      #   class LoginController < ApplicationController
+      #     before_action :require_login, only: %i[index settings logout]
+      #
+      #     def index
+      #     end
+      #
+      #     def settings
+      #     end
+      #
+      #     def logout
+      #     end
+      #   end
+      class LexicallyScopeActionFilter < Cop
+        MSG = '%s not explicitly defined on the controller.'.freeze
+
+        FILTERS = %w[
+          :after_action
+          :append_after_action
+          :append_around_action
+          :append_before_action
+          :around_action
+          :before_action
+          :prepend_after_action
+          :prepend_around_action
+          :prepend_before_action
+          :skip_after_action
+          :skip_around_action
+          :skip_before_action
+          :skip_action_callback
+        ].freeze
+
+        def_node_matcher :only_or_except_filter_methods, <<-PATTERN
+          (send
+            nil?
+            {#{FILTERS.join(' ')}}
+            _
+            (hash
+              (pair
+                (sym {:only :except})
+                $_)))
+        PATTERN
+
+        def on_send(node)
+          methods_node = only_or_except_filter_methods(node)
+          return unless methods_node
+
+          defined_methods = []
```

Comment:
Initializing an empty array and then pushing into it inside an `#each` can be replaced with `#map`. 🙂

Reply:
Umm, I wanted to use the type filter with `#each_child_node`.
Perhaps `#map_child_node` may be necessary.

Reply:
Oops, I noticed that `#each_child_node` will return`Enumerator` when not passing block. I will use this.